lib/tests/test_working_copy.rs

```diff
@@ -210,6 +211,14 @@ fn test_checkout_file_transitions(use_git: bool) {
     wc.check_out(repo.op_id().clone(), None, &right_tree)
         .unwrap();
 
+    // jujutsu doesn't create submodules, so we manually create the
+    // directory here
```

Comment:
Do we have to? Does this test case fail if you don't make any changes to it?

Reply:
yes the test case fails because the submodule object isn't set in the tree builder to be written, should it do that instead?

Reply:
I don't quite understand. Isn't that what lines 166-172 do?

Reply:
Perhaps I misunderstood something, let me recheck the implementation